iOS Client/DevotionsController/DevotionsCoordinator.swift

```diff
@@ -25,13 +26,53 @@ class DevotionsCoordinator: NSObject, Coordinator, StoreSubscriber {
         let devotionsController = MBDevotionsViewController.instantiateFromStoryboard()
         navigationController.pushViewController(devotionsController, animated: true)
         MBStore.sharedStore.subscribe(self)
+        let store = MBStore()
+        let devotions = store.getDevotions()
+        if devotions.count == 0 {
+            store.syncDevotions { syncedDevotions, error in
+                if error != nil {
+                    MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.error))
+                } else if let newDevotions = syncedDevotions {
+                    MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.loaded(data: newDevotions)))
+                }
+            }
+        } else {
+            MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.loaded(data: devotions)))
+        }
+    }
+    
+    func promptForNotifications() {
+        let center = UNUserNotificationCenter.current()
         
-        MBStore().syncDevotions { devotions, error in
-            if error != nil {
-                MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.error))
-            } else if let devotions = devotions {
-                MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.loaded(data: devotions)))
+        center.requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+            if granted {
+                MBStore().syncDevotions { devotions, error in
+                    if error != nil {
+                        print("Could not sync devotions and schedule devotions")
+                        print(error)
+                    } else if devotions != nil {
+                        center.delegate = self
+                        self.scheduleNotifications(devotions: devotions!.map{$0.devotion})
```

Comment:
this code seems out-dated, since the LoadedDevotions no longer contain an embedded MBDevotion